src/lib/core/TLVTags.h

```diff
@@ -163,6 +165,17 @@ constexpr Tag ContextTag(uint8_t tagNum)
     return ProfileTag(Tag::kSpecialTagProfileId, tagNum);
 }
 
+/**
+ * Also allow using enum class values as context tags.  Just using is_enum would
+ * fail for enums that are not enum classes because to_underlying would produce
```

Comment:
Hmm, not sure the comment is correct. For `enum class EC : uint16_t {};`, `std::is_convertible<EC, uint8_t>::value` is still false, so the condition just distinguishes legacy enums from scoped enums (enum classes). To make sure that `to_underlying` doesn't return a type wider than `uint16_t`, something like `std::is_same<std::underlying_type_t<T>, uint8_t>::value` should be used.

Reply:
If the enum is the wrong width, that's no different from the wrong width normal numbers being used, and will be caught by -Wconversion...  But yes, I guess we could catch it more explicitly; I can do that.

Reply:
Thanks!

Reply:
Oh, actually, this doesn't compile, so I'm fine with the previous version if that worked.

Reply:
Yep, I just discovered the "doesn't compile" thing, at least for _some_ compilers.  ;)  Going to revert to what it used to be.

Reply:
But make the comment better.